tests/bootstrap_tests/test_backups.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+from __future__ import with_statement
+
+import itertools
+import os
+import sys
+from mock import patch
+import sgtk
+from sgtk.util import ShotgunPath
+from sgtk.bootstrap.configuration_writer import ConfigurationWriter
+from shutil import copytree, ignore_patterns
+
+from tank.descriptor import Descriptor, descriptor_uri_to_dict, descriptor_dict_to_uri, create_descriptor, constants
+from tank_test.tank_test_base import setUpModule # noqa
+from tank_test.tank_test_base import temp_env_var
+from tank_test.tank_test_base import TankTestBase
+
+
+class TestBackups(TankTestBase):
+    def setUp(self):
+        super(TestBackups, self).setUp()
+
+        self._tmp_bundle_cache = os.path.join(self.tank_temp, "bundle_cache")
+        self._resolver = sgtk.bootstrap.resolver.ConfigurationResolver(
+            plugin_id="backup_tests",
+            bundle_cache_fallback_paths=[self._tmp_bundle_cache]
+        )
+
+    def test_cleanup(self):
+        """
+        Ensures that after a successful update the backup folder created by the
+        update process is properly deleted 
+        """
+        pathHead, pathTail = os.path.split(__file__)
+        core_path=os.path.join(pathHead,"..", "..")
+        temp_test_path=os.path.join(pathHead, "..", "fixtures", "bootstrap_tests", "test_update")
```

Comment:
@jfboismenu I was able to do most of the cleanups and simplifications you suggested above, see latest changes. Except for the copy of tk-core, that had to remain because the 'docs' and 'tests' folders generate file paths too large for Windows to handle and the copy straight out of the tk-core repo fails. So I am just copying it to filter out those 2 folders that are not needed. Let me know if there's a better way of doing this.